packages/credential-governance/src/domain-hierarchy.ts

```diff
@@ -214,35 +216,53 @@ export class DomainHierarchy {
     event: EventFilter;
     contractInterface: utils.Interface;
   }) => {
+    console.group('getDomainsFromLogs');
+
+    let mem = process.memoryUsage().heapUsed;
+    let maxMem = mem;
+    console.log(`max mem ${maxMem / 1e6} mb`);
     const filter = {
       fromBlock: 0,
       toBlock: 'latest',
       address: event.address,
       topics: event.topics || [],
     };
-    // console.log("Inside getAllLogs first map " + (new Date()).toString());
-    const logs = await provider.getLogs(filter);
-    const rawLogs = await pMap(logs, async (log) => {
-      const parsedLog = contractInterface.parseLog(log);
-      return contractInterface.decodeEventLog(parsedLog.name, log.data, log.topics);
-    });
-    
-    // console.log("Inside getAllLogs after first map " + (new Date()).toString());
-    
-    // const rawLogs = logs.map((log) => {
-    //   const parsedLog = contractInterface.parseLog(log);
-    //   /** ethers_v5 Interface.parseLog incorrectly parses log, so have to use lowlevel alternative */
-    //   return contractInterface.decodeEventLog(
-    //     parsedLog.name,
-    //     log.data,
-    //     log.topics
-    //   );
-    // });
-    console.log("Inside getAllLogs second map " + (new Date()).toString());
-    const domains = await pMap(rawLogs, parser);
-    console.log("Inside getAllLogs after second map " + (new Date()).toString());
+    mem = process.memoryUsage().heapUsed;
+    if (mem > maxMem) {
+      maxMem = mem;
+      console.log(`max mem ${maxMem / 1e6} mb`);
+    }
+
+    console.time('domainReader.readName');
+    const domains = await pMap(
+      await provider.getLogs(filter),
+      async (log) => {
+        const parsedLog = contractInterface.parseLog(log);
+        const decoded = contractInterface.decodeEventLog(
+          parsedLog.name,
+          log.data,
+          log.topics
+        );
+        const parsed = await parser(decoded);
+
+        mem = process.memoryUsage().heapUsed;
+        if (mem > maxMem) {
+          maxMem = mem;
+          process.stdout.write(`max mem ${maxMem / 1e6} mb`);
+          process.stdout.clearLine(0);
+          process.stdout.cursorTo(0);
+        }
+
+        return parsed;
+      },
+      { concurrency: 3 }
```

Comment:
@JGiter Is there any reason to set `concurrency` to `3` here, the default concurrency is `infinity` for `pMap`,  I have tested it with different `concurrency` value and the best outcome came with the default value. What do you think?

Reply:
@JGiter I tested the latest changes and it consumes ~250 mbs which seems to be an efficient mem consumption. I think we should merge and release this change, test locally with SSI-Hub and if we don't see any spike then we can push it to dev and later to staging.

I will remove the console logs to get the tests running.

Reply:
Yes lets try this